Task_4/scripts/app.js

```diff
@@ -0,0 +1,102 @@
+const moduleTasks = (function () {
+  const user = "Julia Grib";
+
+  function isValidTypeId(id) {
+    return typeof id === "string";
+  }
+
+  function findTaskById(id) {
+    return tasks.find((task) => task.id === id);
+  }
+
+  function getTask(id) {
+    try {
+      if (!isValidTypeId(id)) {
+        throw new Error(ERRORS.invalidValue);
+      }
+
+      const task = findTaskById(id);
+
+      if (!task) {
+        throw new Error(ERRORS.taskNotFound);
+      }
+
+      return task;
+    } catch (error) {
+      console.error(error);
+
+      return false;
+    }
+  }
+
+  function removeTask(taskId) {
+    try {
+      if (!isValidTypeId(taskId)) {
+        throw new Error(ERRORS.invalidValue);
+      }
+
+      if (!findTaskById(taskId)) {
+        throw new Error(ERRORS.taskNotFound);
+      }
+
+      const initLength = tasks.length;
+      tasks = tasks.filter(({ id, assignee }) => {
+        if (!(taskId === id && assignee === user)) {
+          return true;
+        }
+      });
+
+      if (initLength === tasks.length) {
+        throw new Error(ERRORS.taskNotDel);
+      }
+
+      return true;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  }
+
+  function validateTask(task) {
+    try {
+      if (
+        !(typeof task === "object" && task !== null && !Array.isArray(task))
```

Comment:
remove **!**(typeof task === "object" && task !== null && !Array.isArray(task))
to
(typeof task !== "object" && task === null && Array.isArray(task))